server/channels/store/localcachelayer/file_info_layer.go

```diff
@@ -50,6 +51,41 @@ func (s LocalCacheFileInfoStore) GetForPost(postId string, readFromMaster, inclu
 	return fileInfos, nil
 }
 
+func (s LocalCacheFileInfoStore) GetByIds(ids []string, includeDeleted, allowFromCache bool) ([]*model.FileInfo, error) {
```

Comment:
Only 1/5, but I don't think caching for this is really needed because this data is only fetched when a user specifically views the edit history for a post of theirs which I don't expect to be a very frequent action. Since caching is hard, I'd only expect us to be caching this for something more performance-senstive like getting file infos by post ID since that happens any time a post is loaded.

Reply:
We need cache because if you create a post with some files then edit only the text a few times, the files in each version will be the same. Without cache, the code will fetch the same files each time for each edit history. I need some form of cache- either this or a local map of file ID to file info and figured out this is better.